basic/src/test/java/org/jboss/hal/testsuite/test/runtime/messaging/XaJmsMetricsTestCase.java

```diff
@@ -0,0 +1,161 @@
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.Values;
+import org.wildfly.extras.creaper.core.online.operations.admin.Administration;
+
+@RunWith(Arquillian.class)
+@Category(Shared.class)
+public class XaJmsMetricsTestCase {
+
+    private static final Logger log = LoggerFactory.getLogger(XaJmsMetricsTestCase.class);
+    private static final OnlineManagementClient client = ManagementClientProvider.createOnlineManagementClient();
+    private static final Administration adminOps = new Administration(client);
+    private static final Operations ops = new Operations(client);
+    private static final DeploymentsOperations deployOps = new DeploymentsOperations(client);
+    private static final String ARCHIVE_NAME = "xa-jms-txn", WAR_FILE_EXTENSION = ".war",
+            GENERATE_STATS_URL = "http://" + ConfigUtils.getUrl().getHost() + ":8080/" + ARCHIVE_NAME,
+            SUBSYSTEM_NAME = "messaging-activemq", CONNECTION_FACTORY_NAME = "activemq-ra", SERVER = "server",
+            DEFAULT = "default", POOLED_CONNECTION_FACTORY = "pooled-connection-factory",
+            STATISTICS_ENABLED = "statistics-enabled";
+    private static final Address defaultJmsServerAddress = Address.subsystem(SUBSYSTEM_NAME).and(SERVER, DEFAULT),
+            queueAddress = defaultJmsServerAddress.and("jms-queue", "test"),
+            pooledConnectionFactoryAddress =
+                defaultJmsServerAddress.and(POOLED_CONNECTION_FACTORY, CONNECTION_FACTORY_NAME),
+            pooledConnectionFactoryStatsAddress;
+    private static ModelNodeResult originalFactoryStatsEnabledResult;
+
+    static {
+        Address subsystemRuntimeAddress = ConfigUtils.isDomain() ?
+                Address.host(getDefaultHost()).and(SERVER, MONITORED_SERVER).and(SUBSYSTEM, SUBSYSTEM_NAME)
+                : Address.subsystem(SUBSYSTEM_NAME);
+        pooledConnectionFactoryStatsAddress = subsystemRuntimeAddress.and(SERVER, DEFAULT)
+                .and(POOLED_CONNECTION_FACTORY, CONNECTION_FACTORY_NAME).and("statistics", "pool");
+    }
+
+    @Drone
+    private WebDriver browser;
+
+    @Page
+    private MessagingStatisticsPage page;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        ModelNode entriesNode = new ModelNodeListBuilder(new ModelNode("java:/jms/queue/test")).build();
+        ops.add(queueAddress, Values.of("entries", entriesNode)).assertSuccess();
+        WebArchive war = ShrinkWrap.create(WebArchive.class, ARCHIVE_NAME + WAR_FILE_EXTENSION);
+        war.addPackage(MessagingStatisticsBean.class.getPackage());
+        deployOps.deploy(war);
+
+        originalFactoryStatsEnabledResult = ops.readAttribute(pooledConnectionFactoryAddress, STATISTICS_ENABLED);
+        originalFactoryStatsEnabledResult.assertSuccess();
```

Comment:
I believe you should call `assertDefined()` as after this line you expect it to be defined. WDYT?